ops/main.py

```diff
@@ -384,12 +387,28 @@ def main(charm_class: Type[ops.charm.CharmBase],
     dispatcher = _Dispatcher(charm_dir)
     dispatcher.run_any_legacy_hook()
 
-    metadata = (charm_dir / 'metadata.yaml').read_text()
-    actions_meta = charm_dir / 'actions.yaml'
-    if actions_meta.exists():
-        actions_metadata = actions_meta.read_text()
-    else:
-        actions_metadata = None
+    # Check charmcraft.yaml first and see if it has the metadata data in it.
+    charmcraft_meta = charm_dir / "charmcraft.yaml"
+    if charmcraft_meta.exists():
+        charmcraft_yaml = charmcraft_meta.read_text()
+        charmcraft_metadata: dict[str, Any] = yaml.safe_load(charmcraft_yaml)
+        if any(
+            meta_key in charmcraft_metadata for meta_key in ("name", "summary", "description")
```

Comment:
Is this the behaviour of / test used by charmcraft too? Can you please point to the code there?